src/main/scala/uk/gov/hmrc/ct/validation/TurnoverValidation.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.ct.validation
+
+import org.joda.time.{Days, LocalDate}
+import uk.gov.hmrc.ct.accounts.retriever.AccountsBoxRetriever
+import uk.gov.hmrc.ct.box.ValidatableBox.{commaForThousands, _}
+import uk.gov.hmrc.ct.box._
+import uk.gov.hmrc.ct.box.retriever.{BoxRetriever, FilingAttributesBoxValueRetriever}
+import uk.gov.hmrc.ct.domain.CompanyTypes
+import uk.gov.hmrc.ct.utils.DateImplicits._
+
+trait TurnoverValidation extends Validators {
+
+  self: OptionalIntIdBox =>
+
+  private def daysBetweenDates(start: LocalDate, end: LocalDate): Int = {
+    Days.daysBetween(start, end).getDays + 1
+  }
+
+  protected def isFrs10xHmrcAbridgedReturnWithLongPoA[BR <: AccountsBoxRetriever with FilingAttributesBoxValueRetriever](start: (BR) => StartDate, end: (BR) => EndDate)(boxRetriever: BR): Boolean = {
+    boxRetriever.hmrcFiling().value &&
+      boxRetriever.abridgedFiling().value &&
+      isFRS10x(boxRetriever) &&
+      isLongPoA(boxRetriever, start, end)
+  }
+
+  protected def isFRS10x(boxRetriever: AccountsBoxRetriever): Boolean = {
```

Comment:
these look like utility methods that might be useful in the future and reused, so was wondering if maybe it's worth moving them in a utility class, or perhaps to the box retriever itself (not sure if that's possible though)